### PR TITLE
KPL Typed Config

### PR DESCRIPTION
**Pull request by etspaceman**
Solves #22

### CODE REVIEW

**Comment by etspaceman** [comment]
@markglh this is ready for review

File: src/main/scala/com/weightwatchers/reactive/kinesis/producer/KinesisProducerConfig.scala [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    val wTempDir    = tempDirectory.fold(wRegion)(wRegion.setTempDirectory)
+
+    wTempDir
+  }
+}
+
+object KinesisProducerConfig {
+  def apply() = default
+
+  def default: KinesisProducerConfig = KinesisProducerConfig(
```
Comment (markglh) [inline src/main/scala/com/weightwatchers/reactive/kinesis/producer/KinesisProducerConfig.scala]: I'm a little unsure about having defaults defined inn two places now. Bear in mind that the KPL itself defined some defaults if the values are unspecified, feels like another layer of indirection :/

**Comment by etspaceman** [comment]
Hey Mark - 

Yeah these seem reasonable. I bet I could derive the defaults from the existing typesafe-config loader rather than hardcoding them; it felt awkward to me as well. 

RE your 3rd point, I'll have to think about that and an approach for it. It would make sense for the test to also check for new values against the typed config, so I agree that it would be a valuable addition. 

I might have some time to work on this in the next couple of days, I'll let you know when I have something for you to review.

**Comment by etspaceman** [comment]
@markglh - Just pushed another try. Let me know what you think

**Comment by markglh** [comment]
Thanks @etspaceman, looks good! Just need to fix the Jenkins error (default param issue?):

```
[error] /home/travis/build/WW-Digital/reactive-kinesis/src/main/scala/com/weightwatchers/reactive/kinesis/producer/ProducerConf.scala:31:8: in object ProducerConf, multiple overloaded alternatives of method apply define default arguments.
[error] object ProducerConf {
[error]        ^
[error] one error found
```

**Comment by etspaceman** [comment]
Hrm, odd. Compiled fine on my end. Looking into it

**Comment by etspaceman** [comment]
@markglh I fixed the compile error but now we're getting docker-compose errors on the Jenkins build?

**Comment by markglh** [comment]
waaat - will look into it

**Comment by markglh** [comment]
Ok so travis changed something (it no longer automatically falls back to the sudo env automatically).
 docker & docker-compose are ancient versions in the container env so I've forced it to use the trusty sudo one.

try replacing your travis.yml with this:
https://github.com/WW-Digital/reactive-kinesis/blob/test-build/.travis.yml

We can update it in this Pr instead of raising a new one :)

**Pull request merged**